packages/cozy-harvest-lib/src/helpers/accounts.js

```diff
@@ -68,6 +69,19 @@ export const getTwoFACodeProvider = account => {
   }
 }
 
+export const updateTwoFAState = (account_, { retry, type }) => {
+  const account = clone(account_)
+  let state = retry ? 'TWOFA_NEEDED_RETRY' : 'TWOFA_NEEDED'
+  if (type === 'email') {
+    state += '.EMAIL'
+  } else if (type === 'sms') {
+    state += '.SMS'
+  } else if (type === 'app') {
+    state += '.APP'
+  }
```

Comment:
Nit: if the type is not handled, we have an implicit default case here. Maybe we can handle it explicitly:

```js
const typesToStateFragment = {
  email: 'EMAIL',
  sms: 'SMS',
  app: 'APP'
}

const stateFragment = typesToStateFragment[type]

state += stateFragment ? `.${stateFragment}` : ''
```